### PR TITLE
perf: Optimize lookup for the next deployment by narrowing search

### PR DESCRIPTION
**Pull request by alfrunes**

### CODE REVIEW

File: store/mongo/datastore_mongo.go [hunk 1]
```diff
@@ -2760,6 +2763,41 @@ func (db *DataStoreMongo) FindNewerActiveDeployments(ctx context.Context,
 	return deployments, nil
 }
 
+// FindNewerActiveDeployment finds active deployments which were created
+// after createdAfter for which is either dynamic (requires post processing)
```
Comment (kjaskiewiczz) [inline store/mongo/datastore_mongo.go:2767]: We do not support dynamic deployments here (the code is OK, it's only the comment which is wrong).

File: store/mongo/datastore_mongo.go [hunk 2]
```diff
@@ -463,6 +463,8 @@ const (
 	StorageKeyDeploymentActive              = "active"
 	StorageKeyDeploymentStatus              = "status"
 	StorageKeyDeploymentCreated             = "created"
+	StorageKeyDeploymentDeviceList          = "device_list"
+	StorageKeyDeploymentDynamic             = "dynamic"
```
Comment (kjaskiewiczz) [inline store/mongo/datastore_mongo.go:467]: no dynamic deployments here

**Review by kjaskiewiczz — CHANGES_REQUESTED** [review]
There is no support for dynamic deployments here, so please update the code and commit message. Except for that it's looking good.

**Comment by alfrunes** [comment]
> There is no support for dynamic deployments here, so please update the code and commit message. Except for that it's looking good.

Right, thanks. I actually just backported the commit from enterprise. I'll trim the dynamic deployments references from this PR.

**Comment by alfrunes** [comment]
@mender-test-bot start pipeline --fast

**Review by kjaskiewiczz — APPROVED** [review]

**Pull request merged**